apps/token/src/routes/staking/node/staking-form.tsx

```diff
@@ -331,6 +324,36 @@ export const StakingForm = ({
           )}
         </>
       )}
+      {formState === FormState.Failure && <StakeFailure nodeName={nodeName} />}
+      {formState === FormState.Requested && (
+        <Dialog
+          title="Confirm transaction in wallet"
+          intent={Intent.Warning}
+          open={isDialogVisible}
+          onChange={toggleDialog}
+        >
+          <p>{t('stakingConfirm')}</p>
+        </Dialog>
+      )}
+      {formState === FormState.Pending && (
+        <StakePending
+          action={action}
+          amount={amount}
+          nodeName={nodeName}
+          isDialogVisible={isDialogVisible}
+          toggleDialog={toggleDialog}
+        />
+      )}
+      {formState === FormState.Success && (
+        <StakeSuccess
+          action={action}
+          amount={amount}
+          nodeName={nodeName}
+          removeType={removeType}
+          isDialogVisible={isDialogVisible}
+          toggleDialog={toggleDialog}
+        />
+      )}
```

Comment:
chore (non-blocking): this feels a bit messy could we throw this into it's own component?

Reply:
Can do :)